### PR TITLE
chore: bump ginkgo package

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
chore: bump ginkgo package

add rule for dependabot to ignore ginkgo package, because it needs to be bumped manually.
Bump ginkgo to newest version

**Release note**:
```
Bump ginkgo package
```

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ updates:
       interval: "daily"
     allow:
       - dependency-type: "all"
+    ignore:
+      # Ginkgo update needs a change in automation/build-test.sh.
+      - dependency-name: "github.com/onsi/ginkgo/v2"
```
Comment (0xFelix) [inline .github/dependabot.yml:11]: If we hardcode the gomega version in `automation/build-test.sh` it should be ignored here too.

Comment (ksimon1) [reply .github/dependabot.yml:11]: fixed

**Comment by ksimon1** [comment]
@0xFelix, @akrejcir can you please review?

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**